system/jhtree/jhtree.cpp

```diff
@@ -1845,12 +1870,45 @@ bool CKeyCursor::_gtEqual(IContextLogger *ctx)
         {   
             int rc = node->compareValueAt(recordBuffer, ++nodeKey);
             if (rc <= 0)
-                return true; 
+            {
+                // printf("Still in leaf node at level %u pos %u\n", depth, nodeKey);
+                return true;
+            }
             if (nodeKey < numKeys-1)
             {
                 rc = node->compareValueAt(recordBuffer, numKeys-1);
                 if (rc <= 0)
+                {
+                    // printf("Still in leaf node at level %u\n", depth);
                     lwm = nodeKey+1;
+                }
+            }
+        }
+    }
+    if (!lwm)
+    {
+        while (depth)
+        {
+            depth--;
+            if (depth < maxParentNodes)
+            {
+                node.set(parents[depth]);
```

Comment:
move node.set to line 1905? to avoid the link (+release) if it does not match.  (Still assign it to a temporary.)

Reply:
Ok, have to be careful to set to null properly too...

Reply:
Actually, I suppose you don't really.... lwm remaining 0 means it will get set to root